Sources/PaymentsCore/Networking/APIClient.swift

```diff
@@ -1,84 +1,51 @@
-import Foundation
+import UIKit
 
-public final class APIClient {
-    public typealias CorrelationID = String
+public class APIClient: API {
 
-    private var urlSession: URLSessionProtocol
-    private var environment: Environment
+    private let http: HTTP
+    private let config: CoreConfig
 
-    private var decoder: JSONDecoder = {
-        let decoder = JSONDecoder()
-        decoder.keyDecodingStrategy = .convertFromSnakeCase
-        return decoder
-    }()
+    private let decoder = JSONDecoder()
 
-    public init(environment: Environment) {
-        self.environment = environment
-        self.urlSession = URLSession.shared
+    public convenience init(config: CoreConfig) {
+        self.init(config: config, http: HTTPClient())
     }
 
-    /// For internal use for testing purpose
-    init(urlSession: URLSessionProtocol, environment: Environment) {
-        self.environment = environment
-        self.urlSession = urlSession
+    init(config: CoreConfig, http: HTTP) {
+        self.http = http
+        self.config = config
+        self.decoder.keyDecodingStrategy = .convertFromSnakeCase
     }
 
-    public func fetch<T: APIRequest>(
-        endpoint: T,
-        completion: @escaping (Result<T.ResponseType, NetworkingError>, CorrelationID?) -> Void
-    ) {
-        guard let request = endpoint.toURLRequest(environment: environment) else {
-            completion(.failure(.noURLRequest), nil)
-            return
+    public func send(_ apiRequest: APIRequest, completion: @escaping (HTTPResponse) -> Void) {
+        if let urlRequest = urlRequest(from: apiRequest) {
+            http.send(urlRequest, completion: completion)
```

Comment:
`APIClient` doesn't need to know about the underlying `URLSessionDataTask` to do its job, and receives a single `HTTPResponse` instead of the usual `(URLResponse?, Data?, Error?)` tuple.